LDAR-Sim_inputs.md

```diff
@@ -26,6 +26,263 @@ To support wider use of LDAR-Sim, the University of Calgary and Highwood Emissio
+}
+```
+
+Keep in mind that only 2 parameters have been changed with these parameter files - all other parameters run with the default values. We have attempted to choose representative defaults, but be aware the default values may not be what is appropriate for your use case - it is your responsibility to verify the default parameters are appropriate.
+
+In this example, changing the `n_simulations` key to 3 makes the model run faster and provides a way to test the model without waiting for it to complete 30 simulations. A reasonable workflow with these parameters would be something like running the model in a test configuration with:
+
+```buildoutcfg
+python ldar_sim_main.py parameter_file1.txt parameter_file2.txt
+```
+
+Then, when comfortable understanding the outputs and ready to run the model for longer to get more statistically valid results, run:
+
+```buildoutcfg
+python ldar_sim_main.py parameter_file1.txt
+```
+
+The model will now run for the full 30 simulations because we did not load `parameter_file2.txt` and the `n_simulations` key did not get overwritten to 3. Essentially `parameter_file2.txt` is a run configuration for testing, it is something you could add onto any simulation you like to test out the configuration, then drop it when you are ready to get production results.
+
+To ensure reproducibility, the full set of parameters are written in the output directory so the model run can be reproduced or interrogated.
+
+Keep in mind that the **order matters** - if you change around the order that parameter files are read in, the parameters in the end will be different.
+
+The example above focused on a testing run configuration, but the functionality is designed to be used to enable modularization. For example, parameter files can be modularized into categories like:
+
+- specific asset bases and infrastructure types
+- specific program definitions
+- specific method definitions
+- specific run configurations (e.g., as example above)
+- specific scheduling configurations
+- specific computer systems
+
+For example, running an existing program with an existing collection of methods in a new jurisdiction is as simple as just revising the asset parameter file to the new set of sites, leaving the other parameter files untouched.
+
+While global parameters are straightforward to specify this way, and the above example shows how to do this - a few extra parameters are required to directly specify programs or methods, which are at different levels in the hierarchy.
+
+## LDAR-Sim Parameter Hierarchy
+
+LDAR-Sim uses a 3 level hierarchy of simulations, programs, and methods. To tell LDAR-Sim what level in the hierarchy your parameter file is destined for, you should specify a `parameter_level` parameter that will specify what level your parameter file is aimed at - otherwise LDAR-Sim will interpret it as global.
+
+The `parameter_level` parameter can be one of three values:
+
+- `global`: parameters are aimed at the global level.
```

Comment:
Good question. I don't think so. To do this the programs need to stack by name.

P_ref = default program parameters + P_ref parameters 1 + P_ref parameters 2
P_alt = default program parameters + P_alt parameters  1

P_ref parameters 1, P_ref parameters 2 and P_alt parameters 1 would all be different files. But to do this, you'd need to refer to P_ref by name so P_ref parameters 1 and 2 both stack on that one P_ref program instead of making 2 programs and the sim running with 3 programs.

Methods do this as they have names. Do we want to stack programs by names - I think it would be possible but the names become a whole bunch more important.

Reply:
Set this as issue #91 to see if this is good idea that people want.